### PR TITLE
feat: history management added

### PR DESCRIPTION
**Pull request by vikrantkode**
## Open for reviews
    Upon Login
1. On click on thumbnail .. it will be added to history
2. We can delete from history page by single click on video card and by clear all.
   If not login
       no history page

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -14,9 +15,11 @@ function App() {
       <Route path = "/"element = {<Homepage/>}/> 
       <Route path = "/mock" element = {<Mockman />}/> 
       <Route path = "/videos" element = {<VideoListing />}/> 
+      {/* <Route path = "/videos/:videoId" element = {<VideoListing />}/>  */}
```
Comment (Chiragmodi01) [inline src/App.js:18]: You can remove commented code.

File: src/components/HistoryCard/ViHistoryCard.css [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+.btn-footer:hover {
+    background-color: var(--hover-color);
+
+}
+
+.footer_icons {
+    font-size: 2rem;
+}
+
+.vicons {
```
Comment (Chiragmodi01) [inline src/components/HistoryCard/ViHistoryCard.css:76]: Use better classname, not sure what this reffers to.

File: src/components/HistoryCard/ViHistoryCard.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import axios from "axios"
+
+
+const ViHistoryCard = ({ video }) => {
+  const {state: { encodedToken },} = useAuth();
+    const {dispatch} = useHistory();
+    
+    /** 
+     * @param {*} Removes Single video from history page 
+     */
```
Comment (Chiragmodi01) [inline src/components/HistoryCard/ViHistoryCard.jsx:13]: can be removed

**Review by Chiragmodi01 — COMMENTED** [review]
Overall code structure and ui looks good to me!

File: src/App.js (hunk 1 above)
Comment (sharmayatendra) [inline src/App.js:18]: remove code which is not in use

File: src/components/HistoryCard/ViHistoryCard.css [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+.image img {
+    width: 100%;
+    border-top-left-radius: 0.5rem;
+    border-top-right-radius: 0.5rem;
+}
+
+.title {
+    margin-left: 0.5rem;
+    /* margin-top: 0.5rem; */
```
Comment (sharmayatendra) [inline src/components/HistoryCard/ViHistoryCard.css:29]: remove here as well

File: src/components/HistoryCard/ViHistoryCard.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import "./ViHistoryCard.css";
+import axios from "axios"
+
+
+const ViHistoryCard = ({ video }) => {
+  const {state: { encodedToken },} = useAuth();
+    const {dispatch} = useHistory();
+    
+    /** 
+     * @param {*} Removes Single video from history page
```
Comment (sharmayatendra) [inline src/components/HistoryCard/ViHistoryCard.jsx:12]: good to that you have used JS docs

File: src/components/VideoCard/ViCard.css [hunk 6]
```diff
@@ -78,4 +78,8 @@
 
 .vicons:hover {
     background-color: #f5dadaf3;
```
Comment (sharmayatendra) [inline src/components/VideoCard/ViCard.css:80]: remove hard coded color

**Review by sharmayatendra — APPROVED** [review]
left few suggestions
good work though

File: src/App.js (hunk 1 above)
Comment (vikrantkode) [reply src/App.js:18]: I was trying to make single video page but will make in next functionality. So I kept it

File: src/components/HistoryCard/ViHistoryCard.css (hunk 2 above)
Comment (vikrantkode) [reply src/components/HistoryCard/ViHistoryCard.css:76]: okay

File: src/components/HistoryCard/ViHistoryCard.jsx (hunk 3 above)
Comment (vikrantkode) [reply src/components/HistoryCard/ViHistoryCard.jsx:13]: These are JS docs which you keep to make sure of viewers what functions you are using and what it returns

File: src/components/HistoryCard/ViHistoryCard.jsx (hunk 5 above)
Comment (vikrantkode) [reply src/components/HistoryCard/ViHistoryCard.jsx:12]: ok

File: src/components/VideoCard/ViCard.css (hunk 6 above)
Comment (vikrantkode) [reply src/components/VideoCard/ViCard.css:80]: ok

**Pull request merged**